### PR TITLE
Patch for mlr3#539

### PR DESCRIPTION
**Pull request by mllg**
For https://github.com/mlr-org/mlr3/pull/539.

### CODE REVIEW

File: R/LearnerSurvRpart.R [hunk 1]
```diff
@@ -78,12 +78,8 @@ LearnerSurvRpart = R6Class("LearnerSurvRpart",
...
 
     .predict = function(task) {
-      PredictionSurv$new(
-        task = task,
-        crank = invoke(predict,
-          object = self$model,
-          newdata = task$data(cols = task$feature_names)
-      ))
+      preds = invoke(predict, object = self$model, newdata = task$data(cols = task$feature_names))
+      list(crank = preds)
```
Comment (RaphaelS1) [inline R/LearnerSurvRpart.R:82]: how is the task passed to the `PredictionSurv` in the new method?

File: R/PredictionDataDens.R [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  assert_numeric(pdata$truth, len = n, null.ok = TRUE)
+  assert_numeric(pdata$pdf, len = n, any.missing = FALSE, null.ok = TRUE)
+  assert_numeric(pdata$cdf, len = n, any.missing = FALSE, null.ok = TRUE)
+
+
+  if (!is.null(pdata$distr)) {
+    assert_class(pdata$distr, "VectorDistribution")
+    if (is.null(pdata$crank)) {
+      pdata$crank = unname(pdata$distr$mean())
+    }
```
Comment (RaphaelS1) [inline R/PredictionDataDens.R:20]: this can be deleted, `TaskDens` only has `pdf, cdf` as possible return types

File: R/PredictionDataSurv.R [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#' @export
+is_missing_prediction_data.PredictionDataSurv = function(pdata) { # nolint
+  miss = logical(length(pdata$row_id))
+
+  if (!is.null(pdata$response)) {
+    miss = miss | is.na(self$data$tab$response)
+  }
+
+  if (!is.null(pdata$cran)) {
+    miss = is.na(pdata$crank)
```
Comment (RaphaelS1) [inline R/PredictionDataSurv.R:37]: should this be `miss | is.na(pdata$crank)` ?

File: R/PredictionDataSurv.R [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  result = named_list(elems)
+  result$row_ids = row_ids[ii]
+  for (elem in elems) {
+    result[[elem]] = do.call(c, map(dots, elem))[ii]
+  }
+
+  if ("distr" %in% predict_types) {
+    # FIXME: unsure if this is right?
+    result$distr = do.call(c, map(dots, "distr"))
+  }
```
Comment (RaphaelS1) [inline R/PredictionDataSurv.R:77]: note to self: manually test this

File: R/PredictionDens.R [hunk 5]
```diff
@@ -36,87 +36,39 @@ PredictionDens = R6Class("PredictionDens",
...
-        self$data$tab$pdf = assert_numeric(pdf, len = n, any.missing = FALSE)
+      if (check) {
+        pdata = check_prediction_data(pdata)
       }
 
-      if (!is.null(cdf)) {
-        self$data$tab$cdf = assert_numeric(cdf, len = n, any.missing = FALSE)
-      }
+      self$task_type = "dens"
+      self$man = "mlr3cluster::PredictionDens"
```
Comment (RaphaelS1) [inline R/PredictionDens.R:49]: `mlr3proba::PredictionDens` ?

File: R/PredictionSurv.R [hunk 6]
```diff
@@ -49,61 +49,34 @@ PredictionSurv = R6Class("PredictionSurv",
...
-      }
-
-      if (!is.null(response)) {
-        self$data$tab$response = assert_numeric(response, len = n, any.missing = FALSE)
+      if (check) {
+        pdata = check_prediction_data(pdata)
       }
 
+      self$task_type = "surv"
+      self$man = "mlr3cluster::PredictionSurv"
```
Comment (RaphaelS1) [inline R/PredictionSurv.R:63]: `mlr3proba::PredictionSurv`

File: R/LearnerSurvRpart.R (hunk 1 above)
Comment (mllg) [reply R/LearnerSurvRpart.R:82]: It is not used any more. I'm thinking about removing the argument from the constructor.

Comment (RaphaelS1) [reply R/LearnerSurvRpart.R:82]: How does this work for measures that require training set?

Comment (mllg) [reply R/LearnerSurvRpart.R:82]: RR and BMR have the task and indices and pass it to the measure's score function if required. A manually trained learner has a task stub stored internally which contains the training set.

**Pull request merged**